client/ios/migration-guide.mdx

```diff
@@ -0,0 +1,164 @@
+
+### 1. Package and Class Names
+
+**Old**
+
+```swift
+import RTVIClient
+import RTVIClientIOSDaily
+```
+
+**New**
+
+```swift
+import PipecatClientIOS
+import PipecatClientIOSDaily
+```
+
+### 2. Client and Transport Configuration
+
+Previously, the client constructor accepted parameters like pipeline or endpoint configuration. Now, the configuration is passed explicitly to the `startBotAndConnect()` or `connect()` methods, and the constructor takes only transport options.
+
+**Old**
+
+```swift
+let client = RTVIClient(params: ...)
+```
+
+**New**
+
+```swift
+let pipecatClientOptions = PipecatClientOptions(
+    transport: DailyTransport(),
+    enableMic: true,
+    enableCam: false
+)
+
+let client = PipecatClient(options: pipecatClientOptions)
+```
+
+### 3. Connection Method
+
+The connection process is now explicit and supports modern async handling.
+
+**Old**
+
+```swift
+client.connect()
+```
+
+**New**
+
+```swift
+let startBotParams = APIRequest(endpoint: URL(string: "https://your-server/connect")!)
+client.startBotAndConnect(startBotParams: startBotParams) { result in
+    switch result {
+    case .success(let connectionParams):
+        print("Connected successfully: \(connectionParams)")
+    case .failure(let error):
+        print("Connection failed: \(error)")
+    }
+}
+```
+
+### 4. Function Call Handling
+
+Helpers like `LLMHelper` have been removed. You now register function call handlers directly on the `PipecatClient`.
+
+**Old**
+
+```swift
+let llmHelper = LLMHelper()
+client.registerHelper("llm", llmHelper)
+```
+
+**New**
+
+```swift
+client.registerFunctionCallHandler(functionName: "get_my_current_location") { functionCallData, onResult in
+    let location = await getCurrentLocation()
+    await onResult(location)
+}
+
+client.registerFunctionCallHandler(functionName: "set_restaurant_location") { functionCallData, onResult in
+    handleRestaurantLocation(functionCallData.args)
+    await onResult(.string("success"))
+}
+```
+
+### 5. Pipeline Configuration Initialization
+
+Previously, you could provide a pipeline configuration as part of the `RTVIClient` constructor and it was expected to be in a specific format. Now, if you would like to pass any initial pipeline configurations, you do so as `requestData` added to the endpoint you provide to `connect()`. In both cases, you would need server-side code to parse and apply these settings, but now you can define the structure and what pieces of configuration you want to send.
+
+Check out [this section of docs](../js/api-reference/messages#connection-time-configuration) for an example, complete with server-side code showing how to initialize the pipeline configuration at connection time.
+
+### 6. LLM Context Updates
+
+Previously, context updates were done via helpers. Now, use `appendToContext()`:
+
+```swift
+client.appendToContext(
+    role: "user",
+    content: "Tell me a joke.",
+    run_immediately: true
+)
+```
+
+### 7. Sending Messages and Requests
```

Comment:
Is there a reason you changed this title from "PipelineConfiguration Updates"?

Reply:
Done.